tests/snapshots/FeatureAPI.kt

```diff
@@ -69,40 +69,193 @@ class FeatureAPI : GBIORxWebServices() {
+                    result = 31 * result + anonId.hashCode()       
+                    return result
+                }
+            
+                override fun equals(other: Any?): Boolean {
+                    if (this === other) return true
+                    if (javaClass != other?.javaClass) return false
+            
+                    other as User
+            
+                    if (country != other.country) return false
+                    if (anonId != other.anonId) return false
+            
+                    return true
+                }
+            
+            
+            }
     
-    }
     
+    }
 
     open class Features (
         @SerializedName("ver") val ver: String? = null,  // A version value which should be passed to future calls such that features marked with a "return_if_before" value that comes before ver will not be returned 
         @SerializedName("features") val features: List<FeaturesFeatures>? = null  // A list of feature-variant assignments for the user. Note that if the user is assigned to the "control group" for a feature, no entry will be sent for that feature 
-    ) : Jsonable 
+    ) : Jsonable {
+    
+        override fun toString() = "Features(ver=$ver, features=$features)"
+        
+        override fun hashCode(): Int {
+            var result:Int = ver.hashCode()
+            result = 31 * result + features.hashCode()       
+            return result
+        }
+    
+        override fun equals(other: Any?): Boolean {
+            if (this === other) return true
+            if (javaClass != other?.javaClass) return false
+    
+            other as Features
+    
+            if (ver != other.ver) return false
+            if (features != other.features) return false
+    
+            return true
+        }
+    
+    
+    }
 
     open class FeaturesFeatures (
         @SerializedName("n") val n: String,  // Feature name 
         @SerializedName("r") val r: Boolean? = null,  // FALSE if this feature should not be recorded via analytics events (e.g. we don't need to measure anymore) 
         @SerializedName("v") val v: String? = null,  // Variant for feature. If the variant and feature name are the same, only the feature name will be sent 
         @SerializedName("p") val p: Map<String, Any>? = null,  // Payload for the feature and the variant, if available 
         @SerializedName("l") val l: Map<String, Any>? = null  // Localized resources associated with the variant 
-    ) : Jsonable 
+    ) : Jsonable {
+    
+        override fun toString() = "FeaturesFeatures(n=$n, r=$r, v=$v, p=$p, l=$l)"
+        
+        override fun hashCode(): Int {
+            var result:Int = n.hashCode()
+            result = 31 * result + r.hashCode()       
+            result = 31 * result + v.hashCode()       
+            result = 31 * result + p.hashCode()       
+            result = 31 * result + l.hashCode()       
+            return result
+        }
```

Comment:
Can we not have a simpler version of hashcode Java gives us the `Objects.hash` method which is easier to read too

`Objects.hash(n.hashCode(), r.hashCode() ,v.hashCode(),p.hashCode(),l.hashCode())`

Reply:
This looks like a variant of what the IDE would generate for hashcode and equals.  Looks ok to me.

Reply:
I suppose so, I just cribed the IDE generated one. Its all generated so i'm not really concerned about readability.